src/models/state.js

```diff
@@ -426,6 +426,22 @@ class State extends new Record(DEFAULTS) {
   }
 
   /**
+   * Check whether the selection is empty.
+   *
+   * @return {Boolean}
+   */
+
+  get isEmpty() {
+    const { startKey, endKey, startOffset, endOffset } = this
+
+    if (startKey == endKey) {
+      return (startOffset == endOffset)
+    }
+
+    return this.fragment.text.length == 0
```

Comment:
After checking that the selection is not collapsed, you can probably check if at least one of `startOffset` or `endOffset` equals `0`. It's a cheap check, but a condition for the selection to be empty. It could avoid computing a fragment in most cases.